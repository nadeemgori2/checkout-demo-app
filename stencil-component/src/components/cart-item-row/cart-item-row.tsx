import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'cart-item-row',
  styleUrl: 'cart-item-row.css',
  shadow: true,
})
export class CartItemRow {
  /**
   * Single cart item object to display in the row
   */
  @Prop() item: CartItem;

  render() {
    const { thumbnail, label, qty, price } = this.item;

    return (
      <li class="flex flex-wrap sm:flex-nowrap items-start sm:items-center justify-between gap-4 py-2 min-h-[75px]">
        {/* Product image, title and price */}
        <div class="flex items-start gap-4 flex-1 min-w-0">
          <img src={thumbnail} alt={label} class="w-13 h-13 object-contain rounded-full" />
          <div class="flex-1 min-w-0">
            <a href={label} class="text-[0.95rem] font-medium text-[var(--text-primary)] underline hover:underline whitespace-normal break-words block">
              {label}
            </a>
            <div class="text-xs text-gray-500">
              {price.currency} {price.amount.toFixed(2)}
            </div>

            {/* Mobile layout: show quantity and price stacked */}
            <div class="flex sm:hidden justify-between text-sm mt-1">
              <span class="text-center">{qty} pcs</span>
              <span class="font-semibold text-right">
                {price.currency} {price.amount.toFixed(2)}
              </span>
            </div>
          </div>
        </div>

        {/* Desktop layout: show quantity and price in separate columns */}
        <div class="hidden sm:block text-sm text-center w-10 shrink-0">{qty}</div>
        <div class="hidden sm:block text-sm font-semibold text-right w-24 shrink-0">
          {price.currency} {price.amount.toFixed(2)}
        </div>
      </li>
    );
  }
}
