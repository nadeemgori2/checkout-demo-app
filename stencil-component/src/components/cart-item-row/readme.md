# cart-item-row



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                   | Type       | Default     |
| -------- | --------- | --------------------------------------------- | ---------- | ----------- |
| `item`   | `item`    | Single cart item object to display in the row | `CartItem` | `undefined` |


## Dependencies

### Used by

 - [cart-items](../cart-items)

### Graph
```mermaid
graph TD;
  cart-items --> cart-item-row
  style cart-item-row fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
