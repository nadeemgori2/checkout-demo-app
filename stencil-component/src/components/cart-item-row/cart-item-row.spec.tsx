import { newSpecPage } from '@stencil/core/testing';
import { CartItemRow } from './cart-item-row';
import { h } from '@stencil/core';

describe('cart-item-row', () => {
  const mockItem = {
    id: '001',
    label: 'Test Headphones',
    thumbnail: 'test.jpg',
    qty: 2,
    price: {
      currency: 'SAR',
      amount: 199,
    },
  };

  it('renders with given item data', async () => {
    const page = await newSpecPage({
      components: [CartItemRow],
      template: () => <cart-item-row item={mockItem}></cart-item-row>,
    });
    expect(page.root).toBeTruthy();
  });

  it('renders desktop layout quantity and price blocks', async () => {
    const page = await newSpecPage({
      components: [CartItemRow],
      template: () => <cart-item-row item={mockItem}></cart-item-row>,
    });

    const qty = page.root.shadowRoot.querySelector('.sm\\:block.text-sm.text-center');
    const price = page.root.shadowRoot.querySelector('.sm\\:block.text-sm.font-semibold');

    expect(qty).toBeTruthy();
    expect(price).toBeTruthy();
  });

  it('renders mobile layout stacked quantity and price', async () => {
    const page = await newSpecPage({
      components: [CartItemRow],
      template: () => <cart-item-row item={mockItem}></cart-item-row>,
    });

    const mobileRow = page.root.shadowRoot.querySelector('.sm\\:hidden');
    expect(mobileRow).toBeTruthy();
  });
});
