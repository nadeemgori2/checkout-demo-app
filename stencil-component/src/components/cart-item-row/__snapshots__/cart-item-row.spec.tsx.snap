// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart-item-row renders with given item data 1`] = `
<cart-item-row>
  <template shadowrootmode="open">
    <li class="flex flex-wrap gap-4 items-start justify-between min-h-[75px] py-2 sm:flex-nowrap sm:items-center">
      <div class="flex flex-1 gap-4 items-start min-w-0">
        <img alt="Test Headphones" class="h-13 object-contain rounded-full w-13" src="https://example.com/image.png">
        <div class="flex-1 min-w-0">
          <a class="block break-words font-medium hover:underline text-[0.95rem] text-[var(--text-primary)] underline whitespace-normal" href="Test Headphones">
            Test Headphones
          </a>
          <div class="text-gray-500 text-xs">
            SAR 250.00
          </div>
          <div class="flex justify-between mt-1 sm:hidden text-sm">
            <span class="text-center">
              2 pcs
            </span>
            <span class="font-semibold text-right">
              SAR 250.00
            </span>
          </div>
        </div>
      </div>
      <div class="hidden shrink-0 sm:block text-center text-sm w-10">
        2
      </div>
      <div class="font-semibold hidden shrink-0 sm:block text-right text-sm w-24">
        SAR 250.00
      </div>
    </li>
  </template>
</cart-item-row>
`;
